src/ts_catalog/dimension_partition.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+#ifndef TIMESCALEDB_DIMENSION_PARTITION_H
+#define TIMESCALEDB_DIMENSION_PARTITION_H
+
+#include <postgres.h>
+#include <lib/rbtree.h>
+
+#include "scan_iterator.h"
+
+typedef struct DimensionPartition
+{
+	RBTNode rbtnode;
+	int32 dimension_id;
+	int64 range_start;
+	int64 range_end;
```

Comment:
Yeah, we cache it here since we need it to create chunks after we look up the dimension partition. We don't want to have to look up two partitions during chunk creation.

The cached information is thrown away every time we update the dimension_partition information, so it will be recalculated automatically.

I opted to not embed the Form since I am converting the data nodes from array form to list form